### PR TITLE
Fix: Adding ie9 support

### PR DESCRIPTION
**Pull request by eugeny-dementev**
Adding possibility to make ajax requests in IE9.

### CODE REVIEW

**Comment by pragmadash** [comment]
Thank you @eugeny-dementev but you should fix style errors first. Please take a look at CI build details or just execute `npm test` in the project root.

File: browser/UHR.js [hunk 1]
```diff
@@ -109,7 +109,10 @@ UHR.prototype._doRequest = function (parameters) {
 			requestError = new Error(xhr.statusText || ERROR_CONNECTION);
 			reject(requestError);
 		};
-		xhr.onloadend = function () {
+		xhr.onreadystatechange = function () {
+			if (xhr.readyState !== 4) {
+				return;
+			}
```
Comment (pragmadash) [inline browser/UHR.js]: How will it work with disconnection or timeout? Have you checked these cases?

File: browser/UHR.js [hunk 2]
```diff
@@ -180,7 +183,9 @@ function getStatusObject(xhr) {
 		});
 
 	return {
-		code: xhr.status,
+		// handle IE9 bug:
+		// http://stackoverflow.com/questions/10046972/msie-returns-status-code-of-1223-for-ajax-request
+		code: xhr.status === 1223 ? 204: xhr.status,
```
Comment (pragmadash) [inline browser/UHR.js]: I don't get it. The question by link is about IE8 that is completely unsupported by Catberry. What's the point?

**Comment by pragmadash** [comment]
As far as I concerned, IE9 does not work with cross-domain requests using XHR anyway. 
Have you found a case when it does not work without crosse-domain request?

File: browser/UHR.js (hunk 2 above)
Comment (eugeny-dementev) [inline browser/UHR.js]: Point in bug description. It exist in IE9 to and fixed only in IE10. It's normal if i place google short link here?

**Comment by eugeny-dementev** [comment]
Yes, it does not work with CORS and more, with subdomain to. Only with different port. 
It's really not good, but catberry have partial functionality in IE9 and this maybe good plus in sum.

**Comment by pragmadash** [comment]
Please find my comments on the diff. The second change about the status code does not matter for IE9, right?
It seems like the only change is `onreadystatechange` usage. But I still think that it does not matter either. Have you checked it somehow? And I'm afraid that it can break error handling we need to check it too for sure.

File: browser/UHR.js (hunk 2 above)
Comment (pragmadash) [inline browser/UHR.js]: Feel free to post any links.

File: browser/UHR.js (hunk 1 above)
Comment (eugeny-dementev) [inline browser/UHR.js]: Unfortunately IE9 support only timeout event for this cases.

**Comment by eugeny-dementev** [comment]
Change about status code is really matter. I make this fix when work in IE9 which have all this issues.

File: browser/UHR.js (hunk 1 above)
Comment (pragmadash) [inline browser/UHR.js]: I mean this change could break the existing API in cases when there is no connection or there is a timeout or error. This code just returns in a case when the state is not `4`. I guess, in cases when it's an error it's not `4` and the error won't be handled.

Comment (eugeny-dementev) [inline browser/UHR.js]: Nono, it's exactly READYState. Same as loadend event for real browsers. And it readyState don't have any error codes.

**Comment by pragmadash** [comment]
Okay, the situation related to `xhr.status` is correct as I learnt from the stackoverflow thread.
But I need more time for checking all `onreadystatechange` cases to be sure it does not break anything.

File: browser/UHR.js (hunk 1 above)
Comment (pragmadash) [inline browser/UHR.js]: Fair enough. It should have the same behaviour.

**Comment by pragmadash** [comment]
So, I think it should work and all existed tests are passing. Merging.

**Pull request merged**